### PR TITLE
PMM-6716 Make ComputeResources optional.

### PR DESCRIPTION
**Pull request by askomorokhov**
https://jira.percona.com/browse/PMM-6716

FB: https://github.com/Percona-Lab/pmm-submodules/pull/1177

- https://github.com/percona-platform/dbaas-controller/pull/65
- https://github.com/Percona-Lab/pmm-api-tests/pull/132
- https://github.com/percona/pmm-managed/pull/490
- https://github.com/percona/pmm/pull/594

### CODE REVIEW

File: service/cluster/xtra_db_cluster.go [hunk 1]
```diff
@@ -106,17 +106,21 @@ func (s *XtraDBClusterService) CreateXtraDBCluster(ctx context.Context, req *con
 		Name: req.Name,
 		Size: req.Params.ClusterSize,
 	}
-	params.PXC = &k8sclient.PXC{
-		ComputeResources: &k8sclient.ComputeResources{
-			CPUM:        req.Params.Pxc.ComputeResources.CpuM,
-			MemoryBytes: req.Params.Pxc.ComputeResources.MemoryBytes,
-		},
+	if req.Params.Pxc.ComputeResources.CpuM > 0 && req.Params.Pxc.ComputeResources.MemoryBytes > 0 {
```
Comment (BupycHuk) [inline service/cluster/xtra_db_cluster.go:109]: we should pass one of the values if it's provided

**Review by BupycHuk — CHANGES_REQUESTED** [review]

File: cmd/dbaas-controller/main.go [hunk 2]
```diff
@@ -36,21 +38,20 @@ func main() {
 		panic("dbaas-controller version is not set during build.")
 	}
 
-	logger.SetupGlobal()
```
[the remark this answers is not in the document]
Comment (askomorokhov) [reply cmd/dbaas-controller/main.go:39]: yes, it must be set up after app.Setup:
https://github.com/percona-platform/saas/blob/main/pkg/logger/global.go#L24-L26

File: service/cluster/xtra_db_cluster.go (hunk 1 above)
Comment (askomorokhov) [reply service/cluster/xtra_db_cluster.go]: Fixed.

**Review by BupycHuk — CHANGES_REQUESTED** [review]
looks like we also should fix it here https://github.com/percona-platform/dbaas-controller/blob/2c6a0f5eceda11416adf5bdc429de60bb1d9ad75/service/k8sclient/k8client.go#L371

**Review by artemgavrilov — APPROVED** [review]

**Comment by BupycHuk** [comment]
Please fix this https://github.com/percona-platform/dbaas-controller/pull/65/files#diff-4669ad8e96a2e00e4dab6d2fe999852a567d3f3a907182653bd4acfc754bf36aR371 and then it can be merged

**Review by BupycHuk — APPROVED** [review]

**Review by AlekSi — APPROVED** [review]

**Pull request merged**